src/app/vault/organization-delete.component.ts

```diff
@@ -0,0 +1,98 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+import { CipherService } from '../../services/cipher.service';
+
+import { ApiService } from 'jslib/abstractions/api.service';
+import { CollectionService } from 'jslib/abstractions/collection.service';
+import { CryptoService } from 'jslib/abstractions/crypto.service';
+import { I18nService } from 'jslib/abstractions/i18n.service';
+import { PlatformUtilsService } from 'jslib/abstractions/platformUtils.service';
+import { SearchService } from 'jslib/abstractions/search.service';
+import { UserService } from 'jslib/abstractions/user.service';
+
+import { Organization } from 'jslib/models/domain/organization';
+import { PasswordVerificationRequest } from 'jslib/models/request/passwordVerificationRequest';
+import { CollectionView } from 'jslib/models/view/collectionView';
+
+@Component({
+    selector: 'app-organization-delete',
+    templateUrl: 'organization-delete.component.html',
+})
+export class OrganizationDeleteComponent implements OnInit {
+    @Input() organizationId: string;
+    @Output() onDeletedOrganization = new EventEmitter<Organization>();
+
+    title: string;
+    organizationName: string;
+    organization: Organization;
+    masterPassword: string;
+    formPromise: Promise<any>;
+
+    constructor(private apiService: ApiService, protected i18nService: I18nService,
+        protected platformUtilsService: PlatformUtilsService, private cryptoService: CryptoService,
+        private searchService: SearchService, private userService: UserService,
+        private collectionService: CollectionService, private cipherService: CipherService) { }
+
+    async ngOnInit() {
+        await this.init();
+    }
+
+    async removeAllCiphersFromCollection(collection: CollectionView) {
+        const collectionCiphers = await this.searchService.searchCiphers(
+            null,
+            [c => c.collectionIds != null && c.collectionIds.indexOf(collection.id) > -1],
+            null
+        );
+
+        for (const cipher of collectionCiphers) {
+            await this.cipherService.unshare(cipher);
+        }

```

Comment:
Can't we bulkUnshare instead of unsharing cipher by cipher? Do we make an http request for each cipher?

Reply:
I agree that is is not optimal. But nothing exists yet on stack side (and maybe on bitwarden's protocol) to do it using a bulk request.